File: translations/codenet/deepseek_commented/Go2C/codellama/Code/s414601790.c
#include <stdio.h>

int main() {
    int n;
    char s[100];
    scanf("%d %s", &n, s);

    int max = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++) {
        if (s[i] == 'I') {
            cnt++;
        } else {
            cnt--;
        }

        if (cnt > max) {
            max = cnt;
        }
    }

    printf("%d\n", max);

    return 0;
}

File: translations/codenet/uncommented/Go2P/codellama/Code/s414601790.py

import sys

n, s = sys.stdin.readline().split()
n = int(n)
s = list(s)

max = 0
cnt = 0
for r in s:
    if r == 'I':
        cnt += 1
    else:
        cnt -= 1

    if cnt > max:
        max = cnt

print(max)

